digits/model/views.py

```diff
@@ -189,18 +192,20 @@ def download(job_id, extension):
     elif 'snapshot_epoch' in flask.request.form:
         epoch = float(flask.request.form['snapshot_epoch'])
 
+    # Write the stats of the job to json,
+    # and store in tempfile (for archive)
+    stats = job.get_job_stats_as_json_string(epoch)
+    temp = tempfile.NamedTemporaryFile()
+    temp.write(stats)
+    temp.seek(0)
+
     task = job.train_task()
 
     snapshot_filename = None
-    if epoch == -1 and len(task.snapshots):
-        epoch = task.snapshots[-1][1]
-        snapshot_filename = task.snapshots[-1][0]
-    else:
-        for f, e in task.snapshots:
-            if e == epoch:
-                snapshot_filename = f
-                break
-    if not snapshot_filename:
+
+    try:
```

Comment:
Its not 100% necessary but I figured:

![screenshot from 2016-07-08 18 10 50](https://cloud.githubusercontent.com/assets/2607311/16705174/4b0919ba-4537-11e6-9cc2-6a3d86d9bf75.png)

looked better then:

![screenshot from 2016-07-08 18 12 30](https://cloud.githubusercontent.com/assets/2607311/16705186/7ba74ea2-4537-11e6-9378-783076cf1a5d.png)

In this particular case